hazelcast-client-new/src/test/java/com/hazelcast/client/cache/ClientCacheConfigTest.java

```diff
@@ -170,4 +168,39 @@ public void testGetPreConfiguredCache() {
             Assert.assertNotNull("Cache cannot be retrieved on client: " + i, cache);
         }
     }
+
+    @Test
+    public void createCacheConfigOnAllNodes() {
+        final String CACHE_NAME = "myCache";
+
+        Config config = new Config();
+        CacheSimpleConfig cacheConfig = new CacheSimpleConfig().setName(CACHE_NAME);
+        config.addCacheConfig(cacheConfig);
+
+        // Create servers with configured caches
+        HazelcastInstance server1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance server2 = Hazelcast.newHazelcastInstance(config);
+
+        // Create the hazelcast client instance
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+
+        // Create the client cache manager
+        HazelcastClientCachingProvider cachingProvider = HazelcastClientCachingProvider.createCachingProvider(client);
+        CacheManager cacheManager = cachingProvider.getCacheManager();
+
+        Cache<String, String> cache = cacheManager.getCache(CACHE_NAME);
+
+        assertNotNull(cache);
+
+        // First attempt to use the cache will trigger to create its record store
+        for (int i = 0; i < 10; i++) {
```

Comment:
I understand the behaviour. The test is called 'createCacheConfigOnAllNodes'. 

I think it makes sense to validate explicitly that the config has been created on all nodes using an operation. This way you don't rely on side effects to see your system really behaves like expected. Please have a look at the test example I provided; isn't that very explicit?